src/main/java/other/BitcoinService.java

```diff
@@ -17,28 +18,30 @@
     private static final RequestSpecification SPECIFICATION = new RequestSpecBuilder().setBaseUri(baseUri).build();
     private ObjectMapper objectMapper = new ObjectMapper();
 
-    public void getListOfAverages() {
-
+    public ArrayList getListOfAverages() {
 
         List<BitcoinDay> bitcoinDay = null;
+        ArrayList average = new ArrayList();
 
             bitcoinDay = readValue(HttpRequests.executeGet(SPECIFICATION,"indices/global/history/BTCUSD?period=alltime", 200), BitcoinDay.class);
 
         for (int i = 0; i < 30; i++) {
-            System.out.println("Average" + " " + (i+1) + " " + "=" + " " + bitcoinDay.get(i).getAverage());
+            average.add(i, bitcoinDay.get(i).getAverage());
         }
+
+        return average;
     }
 
-        private List<BitcoinDay> readValue(String response, Class valueType) {
+    private List<BitcoinDay> readValue(String response, Class valueType) {
 
-            List<BitcoinDay> bitcoinDay = null;
+        List<BitcoinDay> bitcoinDay = null;
 
-            try {
-                bitcoinDay = objectMapper.readValue(response, objectMapper.getTypeFactory().constructCollectionType(List.class, valueType));
-            } catch (IOException e) {
-                e.printStackTrace();
+        try {
+            bitcoinDay = objectMapper.readValue(response, objectMapper.getTypeFactory().constructCollectionType(List.class, valueType));
+        }   catch (IOException e) {
+            e.printStackTrace();
```

Comment:
I would encapsulate this exception handling in a separate method

Reply:
On our last offline session, you asked me to write custom readValue method in order to handle exception there(and not in getListOfBitcoinAverages as it was before). That's what i did.